### PR TITLE
Add Tashkeela dataset

### PR DESCRIPTION
**Pull request by zaidalyafeai**
Arabic Vocalized Words Dataset.

### CODE REVIEW

File: datasets/tashkeela/tashkeela.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        """BuilderConfig for Tashkeela.
+
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(TashkeelaConfig, self).__init__(version=datasets.Version("1.0.0", ""), **kwargs)
+
+
+class Metrec(datasets.GeneratorBasedBuilder):
+    """Metrec dataset."""
```
Comment (lhoestq) [inline datasets/tashkeela/tashkeela.py:61]:
```suggestion
class Tashkeela(datasets.GeneratorBasedBuilder):
    """Tashkeela dataset."""
```

File: datasets/tashkeela/dataset_infos.json [hunk 2]
```diff
@@ -0,0 +1 @@
...y vocalized words mainly97 books from classical and modern Arabic language.\n", "citation": "@article{zerrouki2017tashkeela,\n  title={Tashkeela: Novel corpus of Arabic vocalized texts, data for auto-diacritization systems},\n  author={Zerrouki, Taha and Balla, Amar},\n  journal={Data in brief},\n  volume={11},\n  pages={147},\n  year={2017},\n  publisher={Elsevier}\n}\n", "homepage": "https://github.com/zaidalyafeai/Tashkeela", "license": "", "features": {"text": {"dtype": "string", "id": null, "_type": "Value"}, "book": {"dtype": "string", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "metrec", "config_name": "plain_text", "version": {"version_str": "1.0.0", "description": "", "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 1081110249, "num_examples": 97, "dataset_name": "metrec"}}, "download_checksums": {"https://sourceforge.net/projects/tashkeela/files/latest/download": {"num_bytes": 183393530, "checksum": "bfa2353cabaf4f1e8962f411201e4a4f936a70fa4981758d8994b35d7cf37b33"}}, "download_size": 183393530, "post_processing_size": null, "dataset_size": 1081110249, "size_in_bytes": 1264503779}}
```
Comment (lhoestq) [inline datasets/tashkeela/dataset_infos.json:1]:
```suggestion
{"plain_text": {"description": "Arabic vocalized texts.\nit contains 75 million of fully vocalized words mainly97 books from classical and modern Arabic language.\n", "citation": "@article{zerrouki2017tashkeela,\n  title={Tashkeela: Novel corpus of Arabic vocalized texts, data for auto-diacritization systems},\n  author={Zerrouki, Taha and Balla, Amar},\n  journal={Data in brief},\n  volume={11},\n  pages={147},\n  year={2017},\n  publisher={Elsevier}\n}\n", "homepage": "https://github.com/zaidalyafeai/Tashkeela", "license": "", "features": {"text": {"dtype": "string", "id": null, "_type": "Value"}, "book": {"dtype": "string", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "tashkeela", "config_name": "plain_text", "version": {"version_str": "1.0.0", "description": "", "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 1081110249, "num_examples": 97, "dataset_name": "tashkeela"}}, "download_checksums": {"https://sourceforge.net/projects/tashkeela/files/latest/download": {"num_bytes": 183393530, "checksum": "bfa2353cabaf4f1e8962f411201e4a4f936a70fa4981758d8994b35d7cf37b33"}}, "download_size": 183393530, "post_processing_size": null, "dataset_size": 1081110249, "size_in_bytes": 1264503779}}
```
`metrec` -> `tashkeela`

File: datasets/tashkeela/README.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+source_datasets:
+- original
+task_categories:
+- sequence-modeling
+task_ids:
+- language-modeling
+- other-diacritics-prediction
+---
+
+# Dataset Card for MetRec
```
Comment (lhoestq) [inline datasets/tashkeela/README.md:23]:
```suggestion
# Dataset Card for Tashkeela
```

File: datasets/tashkeela/tashkeela.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                gen_kwargs={
+                    "directory": os.path.join(arch_path, "Tashkeela-arabic-diacritized-text-utf8-0.3/texts.txt")
+                },
+            ),
+        ]
+
+    def _generate_examples(self, directory):
+        """Generate examples."""
+
+        for id_, file_name in enumerate(glob.glob(f"{directory}/**.txt")):
```
Comment (lhoestq) [inline datasets/tashkeela/tashkeela.py:98]:
```suggestion
        for id_, file_name in enumerate(sorted(glob.glob(os.path.join(directory, "**.txt"))):
```
we need to use `sorted` to make it deterministic no matter the environment

**Review by lhoestq — APPROVED** [review]
Looks all good thanks !

**Comment by zaidalyafeai** [comment]
Sorry @lhoestq for the trouble, sometime I forget to change the names :/

File: datasets/tashkeela/tashkeela.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                gen_kwargs={
+                    "directory": os.path.join(arch_path, "Tashkeela-arabic-diacritized-text-utf8-0.3/texts.txt")
+                },
+            ),
+        ]
+
+    def _generate_examples(self, directory):
+        """Generate examples."""
+
+        for id_, file_name in enumerate(sorted(glob.glob(os.path.join(directory, "**.txt"))):
```
Comment (lhoestq) [inline datasets/tashkeela/tashkeela.py:98]:
```suggestion
        for id_, file_name in enumerate(sorted(glob.glob(os.path.join(directory, "**.txt")))):
```
missing one parenthesis

**Comment by lhoestq** [comment]
> Sorry @lhoestq for the trouble, sometime I forget to change the names :/

haha it's ok ;)

**Pull request merged**